File: daily_problems/2024/07/0722/personal_submission/cf1172b_yawn_sean.cpp
#include <bits/stdc++.h>
#pragma GCC optimize("O3")
using namespace std;

signed main()
{
    ios_base::sync_with_stdio(false);
    cin.tie(0);
    cout.tie(0);

    int n;
    cin >> n;

    vector<int> cnt(n, 1);
    int ans = n, mod = 998244353;

    for (int i = 0; i < n - 1; i ++) {
        int u, v;
        cin >> u >> v;
        u --, v --;
        ans = 1ll * ans * cnt[u] % mod * cnt[v] % mod;
        cnt[u] ++, cnt[v] ++;
    }
    cout << ans;

    return 0;
}

File: daily_problems/2024/07/0722/personal_submission/cf1172b_yawn_sean.py
# Submission Link: https://codeforces.com/contest/1172/submission/271839890
def main():
    n = II()
    cnt = [1] * n
    mod = 998244353

    ans = n
    for _ in range(n - 1):
        u, v = GMI()
        ans = ans * cnt[u] % mod * cnt[v] % mod
        cnt[u] += 1
        cnt[v] += 1

    print(ans)